src/Microsoft.Extensions.Caching.Memory/MemoryCacheServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for setting up memory cache related services in an <see cref="IServiceCollection" />.
+    /// </summary>
+    public static class MemoryCacheServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds memory caching services to the specified <see cref="IServiceCollection" />.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        public static void AddMemoryCache(this IServiceCollection services)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            services.AddOptions();
+            services.TryAdd(ServiceDescriptor.Transient<IDistributedCache, MemoryDistributedCache>());
```

Comment:
This still seems wrong to me. There's no easy way to add just an implementation of MemoryCache without MemoryDistributedCache? Just because both happen to store things in memory that doesn't seem like they both belong here. What if we add more types of caches in the future? Does this method always register all of them with defaults?